src/model/AreaDataSource.ts

```diff
@@ -287,6 +287,10 @@ export default class AreaDataSource extends MongoDataSource<AreaType> {
       })
     }
     const ancestorsCSV = ancestors.join(',')
-    return await this.findDescendantsByPath(ancestorsCSV)
+    const [leafAreas, nonLeafAreas] = await Promise.all([
+      this.findDescendantsByPath(ancestorsCSV, true),
+      this.findDescendantsByPath(ancestorsCSV, false)
+    ])
```

Comment:
This works, but couldn't you just remove the boolean from the method entirely? When would you only want leaf vs non-leaf?

Reply:
I could do it that way, but this way does have a little bit of a performance benefit, depending on the ratio of leaf to non-leaf areas.

I will say, locally leaf areas account for ~90% of all areas in Colorado and California (which i tested). So I'm concerned that you'll see a huge slow down after this merge, but at least for now, better to get the right data slowly, than get not nearly enough data.

I'm going to go ahead and merge, and i'll do my best to work on [speeding up performance properly](https://github.com/OpenBeta/openbeta-graphql/issues/458) in the next couple weeks.